wandb/sdk/wandb_run.py

```diff
@@ -3701,6 +3705,54 @@ def pb_to_dict(
                 logger.error("Error getting system metrics: %s", e)
         return {}
 
+    @property
+    @_run_decorator._noop_on_finish()
+    def _metadata(self) -> Metadata | None:
```

Comment:
you want to make sure that it works correctly on:
- disabled mode
- attach mode

Reply:
ok, made it read-only from attached runs, I think it's cleaner

Reply:
yeah, the problem is that metadata is a mix of system stuff and some user stuff, but maybe limiting it to the main node is a reasonable compromise. This part is hard with merging different behaviors